core/src/main/java/org/javarosa/core/model/actions/SetValueAction.java

```diff
@@ -68,37 +68,56 @@ public void processAction(FormDef model, TreeReference contextRef) {
         //reasonably
         EvaluationContext context = new EvaluationContext(model.getEvaluationContext(), qualifiedReference);
 
-        Object result;
-
-        if (explicitValue != null) {
-            result = explicitValue;
-        } else {
-            result = XPathFuncExpr.unpack(value.eval(model.getMainInstance(), context));
-        }
-        
         String failMessage = "Target of TreeReference " + target.toString(true) + " could not be resolved!";
-        
-        //If the reference has predicates we need to qualify it further.
+
         if(qualifiedReference.hasPredicates()) {
+            //CTS: in theory these predicates could contain logic which breaks if the qualified ref
+            //contains unbound repeats (IE: nested repeats).
             Vector<TreeReference> references = context.expandReference(qualifiedReference);
-            if(references.size() == 0) {
+            if (references.size() == 0) {
+                //If after finding our concrete reference it is a template, this action is outside of the
+                //scope of the current target, so we can leave.
+                if(model.getMainInstance().hasTemplatePath(target)) {
+                    return;
+                }
                 throw new NullPointerException(failMessage);
-            }
-            else if(references.size() > 1) {
-                //The spec for single value binding says we should pick the first one,
-                //but that's not how we do things to prevent errors. We're going to fail
-                //the same way a nodeset does
+            } else if (references.size() > 1) {
                 throw new XPathTypeMismatchException("XPath nodeset has more than one node [" + XPathNodeset.printNodeContents(references) + "]; Actions can only target a single node reference. Refine path expression to match only one node.");
-            }
-            else {
+            } else {
                 qualifiedReference = references.elementAt(0);
             }
         }
 
+        //If after finding our concrete reference it is a template, this action is outside of the
+        //scope of the current target, so we can leave.
+        if(qualifiedReference.isTemplateRef()) {
+            return;
+        }
```

Comment:
This conditional is never satisfied by any tests in the test suite. Since this is in the depths of some pretty complex and uses the newly introduced `isTemplateRef`, I think it is important to write a test case that triggers a `return` here.